internal/impl/postgresql/input_postgrecdc.go

```diff
@@ -0,0 +1,562 @@
+		dbPort                  int
+		dbHost                  string
+		dbSchema                string
+		dbUser                  string
+		dbPassword              string
+		dbSlotName              string
+		temporarySlot           bool
+		tlsSetting              string
+		tables                  []string
+		streamSnapshot          bool
+		snapshotMemSafetyFactor float64
+		decodingPlugin          string
+		pgConnOptions           string
+		streamUncomited         bool
+		snapshotBatchSize       int
+		checkpointLimit         int
+		batching                service.BatchPolicy
+	)
+
+	dbSchema, err = conf.FieldString("schema")
+	if err != nil {
+		return nil, err
+	}
+
+	dbSlotName, err = conf.FieldString("slot_name")
+	if err != nil {
+		return nil, err
+	}
+
+	temporarySlot, err = conf.FieldBool("temporary_slot")
+	if err != nil {
+		return nil, err
+	}
+
+	if dbSlotName == "" {
+		dbSlotName = randomSlotName
+	}
+
+	dbPassword, err = conf.FieldString("password")
+	if err != nil {
+		return nil, err
+	}
+
+	dbUser, err = conf.FieldString("user")
+	if err != nil {
+		return nil, err
+	}
+
+	tlsSetting, err = conf.FieldString("tls")
+	if err != nil {
+		return nil, err
+	}
+
+	dbName, err = conf.FieldString("database")
+	if err != nil {
+		return nil, err
+	}
+
+	dbHost, err = conf.FieldString("host")
+	if err != nil {
+		return nil, err
+	}
+
+	dbPort, err = conf.FieldInt("port")
+	if err != nil {
+		return nil, err
+	}
+
+	tables, err = conf.FieldStringList("tables")
+	if err != nil {
+		return nil, err
+	}
+
+	if checkpointLimit, err = conf.FieldInt("checkpoint_limit"); err != nil {
+		return nil, err
+	}
+
+	streamSnapshot, err = conf.FieldBool("stream_snapshot")
+	if err != nil {
+		return nil, err
+	}
+
+	streamUncomited, err = conf.FieldBool("stream_uncomited")
+	if err != nil {
+		return nil, err
+	}
+
+	decodingPlugin, err = conf.FieldString("decoding_plugin")
+	if err != nil {
+		return nil, err
+	}
+
+	snapshotMemSafetyFactor, err = conf.FieldFloat("snapshot_memory_safety_factor")
+	if err != nil {
+		return nil, err
+	}
+
+	snapshotBatchSize, err = conf.FieldInt("snapshot_batch_size")
+	if err != nil {
+		return nil, err
+	}
+
+	if pgConnOptions, err = conf.FieldString("pg_conn_options"); err != nil {
+		return nil, err
+	}
+
+	if pgConnOptions != "" {
+		pgConnOptions = "options=" + pgConnOptions
+	}
+
+	if batching, err = conf.FieldBatchPolicy("batching"); err != nil {
+		return nil, err
+	} else if batching.IsNoop() {
+		batching.Count = 1
+	}
+
+	pgconnConfig := pgconn.Config{
+		Host:     dbHost,
+		Port:     uint16(dbPort),
+		Database: dbName,
+		User:     dbUser,
+		TLSConfig: &tls.Config{
+			InsecureSkipVerify: true,
+		},
+		Password: dbPassword,
+	}
+
+	if tlsSetting == "none" {
+		pgconnConfig.TLSConfig = nil
+	}
+
+	snapsotMetrics := mgr.Metrics().NewGauge("snapshot_progress", "table")
+	replicationLag := mgr.Metrics().NewGauge("replication_lag_bytes")
+	snapshotMessageRate := mgr.Metrics().NewGauge("snapshot_message_rate")
```

Comment:
Can we prefix these with `postgres_cdc` or something? So it's more clear which input it's for if you're looking at stats of many different instances.

Reply:
Also for the message rate - I think it's preferred to just use a counter and let a stats engine like influxdb or prometheus compute the rate.